iceoryx_utils/include/iceoryx_utils/internal/cxx/function_ref.inl

```diff
@@ -34,6 +34,21 @@ inline function_ref<ReturnType(ArgTypes...)>::function_ref(CallableType&& callab
 {
 }
 
+template <class ReturnType, class... ArgTypes>
+inline function_ref<ReturnType(ArgTypes...)>::function_ref(ReturnType (*function)(ArgTypes...))
```

Comment:
I'd suggest to avoid an additional overload here. It should be possible with `std::is_object` to determine what kind of callable was passed to the c'tor. However, I suppose `std::addressof` to be [implemented with std::is_object](https://en.cppreference.com/w/cpp/memory/addressof).

Reply:
Well, for function pointer we cannot use ``std::addressof`` it seems. It was not working at all for function pointers before (but please double check). Personally I consider using an almost unrestricted version of universal reference dangerous, it can essentially take anything. And there is still an issue that it can be used with e.g.method pointers and I am not sure that anything that is happening inside the function is fully portable in this case alone.

If the behavior is (even slighly) different for specific types, there is no way around an overload.

Reply:
I will create an example which fails with the original later.